### PR TITLE
Fix install on OSX

### PR DESCRIPTION
**Pull request by zloidemon**
Patch based on https://github.com/tarantool/pg/pull/13

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -43,6 +43,11 @@ include_directories(${TARANTOOL_INCLUDE_DIRS})
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
 set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -Wall -Wextra")
 
+if(${CMAKE_SYSTEM_NAME} STREQUAL "Darwin")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -undefined dynamic_lookup")
+    set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -undefined dynamic_lookup")
```
Comment (bigbes) [inline CMakeLists.txt]: Please, replace `CMAKE_C_FLAGS*` with next directive:
```
set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -undefined suppress -flat_namespace")
```

Comment (zloidemon) [inline CMakeLists.txt]: Pushed.

Comment (romanhabibov) [reply CMakeLists.txt]: > Please, replace `CMAKE_C_FLAGS*` with next directive:
> 
> ```
> set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -undefined suppress -flat_namespace")
> ```

It is already added with 496362b1cd7469630d1ca6d2d1c3ecc446c7bd49

**Comment by Totktonada** [comment]
As Roman pointed, the fix is already pushed in 496362b1cd7469630d1ca6d2d1c3ecc446c7bd49. I don't know, what check is more appropriate here: `if (APPLE) <...> endif()` or `if (${CMAKE_SYSTEM_NAME} STREQUAL "Darwin") <...> endif()`, but it seems, both are work: at least I don't know counter-examples.

**Pull request closed without merging**